src/modules/ApprovalVotingModule.sol

```diff
@@ -271,8 +247,13 @@ contract ApprovalVotingModule is VotingModule {
                 }
             }
 
-            // Break loop if `budgetAmount` is exceeded
             if (settings.budgetAmount != 0) {
+                // If `budgetToken` is not ETH and `option.budgetAmount` is not zero, add amount to `totalValue`
+                if (settings.budgetToken != address(0)) {
+                    if (option.budgetTokensSpent != 0) totalValue += option.budgetTokensSpent;
+                }
+
+                // Break loop if `budgetAmount` is exceeded
                 if (totalValue > settings.budgetAmount) break;
             }
```

Comment:
There's probably a better way to write this logic, but I'm not sure I get your suggestion. Where would you add the `else`?

Since comments are not very clear here, let me clarify what happens with the `totalValue` logic:

- In `n` for loop (L232)
  - if token is ETH, we add `option.value` to `totalValue`, and
  - if `totalValue` exceeds budget, we break the `n` loop

- in `i` for loop (L250)
  - if token is not ETH, we add `option.value` to `totalValue`
  - if `totalValue` exceeds budget, **both from ETH and non-ETH tokens**, we break the `i` loop

This last step is fundamental to ensure that once an option exceeds the budget with one of its transactions, that option is not included and `_formatExecuteParams` returns the options until that point. Which is the intended design for this module.

Reply:
I gotcha. I was missing that the length was incrementing later so the outputted arrays would be capped by breaking the `i` loop, even if the params were added.

Nevermind my comment, this looks good.